p2panda-net/src/lib.rs

```diff
@@ -8,15 +8,36 @@ pub mod network;
 mod protocols;
 mod sync;
 
+use std::fmt::{Debug, Display};
+
 pub use addrs::{NodeAddress, RelayUrl};
 pub use config::Config;
 pub use message::{FromBytes, ToBytes};
 pub use network::{Network, NetworkBuilder, RelayMode};
+use p2panda_core::Hash;
 pub use protocols::ProtocolHandler;
 pub use tokio_util::task::AbortOnDropHandle;
 
 #[cfg(feature = "log-sync")]
 pub use p2panda_sync::log_sync::LogSyncProtocol;
 
 pub type NetworkId = [u8; 32];
-pub type TopicId = [u8; 32];
+
+/// Topics are identified by a network-unique id encoded as a `[u8; 32]`.
+/// Topic ids are announced on the network and used to identify peers with
+/// similar interests. Once identified, peers join a gossip overlay and, if
+/// a sync protocol has been provided, attempt to synchronize past state.
```

Comment:
Not sure if we've settled on "live mode" or if we want to refer to this in some other way.

```suggestion
/// Topic ids are announced on the network and used to identify peers with
/// overlapping interests. Once identified, peers join a gossip overlay and, if
/// a sync protocol has been provided, attempt to synchronize past state before entering "live mode".
```